experimental/handoff-library/csharp_dotnetcore/Microsoft.Bot.Builder.Handoff/HandoffRequest.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Handoff;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+using Microsoft.Rest;
+
+namespace Microsoft.Bot.Builder
+{
+    /// <summary>
+    /// The interface allows callers to track the completion of the handoff request.
+    /// </summary>
+    public class HandoffRequest : IHandoffRequest
+    {
+        private readonly string conversationId;
+        private readonly IConversations conversations;
+
+        public HandoffRequest(string conversationId, IConversations conversations)
+        {
```

Comment:
argument check here, and throw on both parameters. 

Our coding standard is to validate all arguments of public functions.